lib/OpenQA/Worker/Job.pm

```diff
@@ -981,22 +984,19 @@ sub _upload_asset {
     log_info("Uploading $filename using multiple chunks", channels => \@channels_worker_only, default => 1);
 
     $ua->upload->once(
-        'upload_local.prepare' => sub {
-            my ($self) = @_;
+        'upload_local.prepare' => sub ($upload, @args) {
             log_info("$filename: local upload (no chunks needed)", channels => \@channels_worker_only, default => 1);
             chmod 0644, $file;
         });
     $ua->upload->once(
-        'upload_chunk.prepare' => sub {
-            my ($self, $pieces) = @_;
+        'upload_chunk.prepare' => sub ($upload, $pieces, @args) {
             log_info("$filename: " . $pieces->size() . " chunks",   channels => \@channels_worker_only, default => 1);
             log_info("$filename: chunks of $chunk_size bytes each", channels => \@channels_worker_only, default => 1);
         });
     my $t_start;
     $ua->upload->on('upload_chunk.start' => sub { $t_start = time() });
     $ua->upload->on(
-        'upload_chunk.finish' => sub {
-            my ($self, $piece) = @_;
+        'upload_chunk.finish' => sub ($upload, $piece, @args) {
```

Comment:
This event doesn't actually have more arguments, what is `@args` supposed to do here?